File: frontend/src/types/player.ts
export interface Player {
  name: string
  realm: Realm
  cultivation: number
  spiritStones: number
  stats: PlayerStats
  currentHp: number
  maxHp: number
  currentMp: number
  maxMp: number
  sect: Sect | null
  location: string
  artifacts: Artifact[]
  equippedArtifact: Artifact | null
  skills: LearnedSkill[]
  mainSkillId: string | null
  activeQuests: string[]
  completedQuests: string[]
  karma: number
  playTime: number
}

export interface PlayerStats {
  hp: number
  mp: number
  attack: number
  defense: number
  speed: number
  wisdom: number
  luck: number
}

export type Realm =
  | '练气期'
  | '筑基期'
  | '金丹期'
  | '元婴期'
  | '化神期'
  | '合体期'
  | '渡劫期'

export type Sect = '峨眉派' | '青城派' | '昆仑派' | '血河教'

export type Element = '金' | '木' | '水' | '火' | '土'

export interface Artifact {
  id: string
  name: string
  artifactType: ArtifactType
  element: Element
  power: number
  speed: number
  level: number
  maxLevel: number
  description: string
  quality: Quality
}

export type ArtifactType = '飞剑' | '防御' | '攻击' | '辅助' | '特殊'

export type Quality =
  | 'common'
  | 'uncommon'
  | 'rare'
  | 'epic'
  | 'legendary'
  | 'mythic'
  | 'divine'

export interface LearnedSkill {
  id: string
  name: string
  level: number
  maxLevel: number
  description: string
}

File: game/player.py
class Player:
    def __init__(self, name="主角", level=1, realm="练气期"):
        self.name = name
        self.level = level
        self.realm = realm
        self.hp = 100
        self.max_hp = 100
        self.mp = 50
        self.max_mp = 50
        self.attack = 20
        self.defense = 10
        self.speed = 15
        self.skills = ["剑斩", "火球术"]
        self.inventory = []
    
    def to_dict(self):
        return {
            "name": self.name,
            "level": self.level,
            "realm": self.realm,
            "hp": self.hp,
            "max_hp": self.max_hp,
            "mp": self.mp,
            "max_mp": self.max_mp,
            "attack": self.attack,
            "defense": self.defense,
            "speed": self.speed,
            "skills": self.skills,
            "inventory": self.inventory
        }
    
    def from_dict(self, data):
        self.name = data.get("name", "主角")
        self.level = data.get("level", 1)
        self.realm = data.get("realm", "练气期")
        self.hp = data.get("hp", 100)
        self.max_hp = data.get("max_hp", 100)
        self.mp = data.get("mp", 50)
        self.max_mp = data.get("max_mp", 50)
        self.attack = data.get("attack", 20)
        self.defense = data.get("defense", 10)
        self.speed = data.get("speed", 15)
        self.skills = data.get("skills", ["剑斩", "火球术"])
        self.inventory = data.get("inventory", [])
        return self
    
    def attack_enemy(self, enemy):
        damage = max(1, self.attack - enemy.defense)
        enemy.hp = max(0, enemy.hp - damage)
        return damage
    
    def use_skill(self, skill_name, enemy):
        if skill_name == "剑斩":
            damage = max(1, self.attack * 1.5 - enemy.defense)
            enemy.hp = max(0, enemy.hp - damage)
            return damage
        elif skill_name == "火球术":
            if self.mp >= 10:
                self.mp -= 10
                damage = max(1, self.attack * 2 - enemy.defense)
                enemy.hp = max(0, enemy.hp - damage)
                return damage
            else:
                return 0
        return 0
